jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/AddDataCaseFileInstanceCommand.java

```diff
@@ -81,6 +82,16 @@ public Void execute(Context context) {
         // apply authorization
         authorizationManager.checkDataAuthorization(caseFile.getCaseId(), caseFile, parameters.keySet());
         
+        // check read only variables, due to dependencies between firexxxCaseData methods, we need this logic here, we cannot reuse VariableScopeInstance.setVariable
```

Comment:
Yes, after trying to modify original code to call setVariable instance and centralize some logic, some test were failing because the trigger call needs to be invoked at a very specific moment of time (or the case will be added twice), so I was adding that comment to avoid the temptation to do the same change in the future